ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/privacy/methods/eea/EeaGetTransactionReceipt.java

```diff
@@ -92,18 +92,19 @@ public JsonRpcResponse response(final JsonRpcRequest request) {
     PrivateTransaction privateTransaction;
     String privacyGroupId;
     try {
-      ReceiveResponse receiveResponse = getReceiveResponseFromEnclave(transaction, publicKey);
+      final ReceiveResponse receiveResponse = getReceiveResponseFromEnclave(transaction, publicKey);
       LOG.trace("Received transaction information from Enclave");
 
       final BytesValueRLPInput bytesValueRLPInput =
           new BytesValueRLPInput(BytesValues.fromBase64(receiveResponse.getPayload()), false);
 
       privateTransaction = PrivateTransaction.readFrom(bytesValueRLPInput);
       privacyGroupId = receiveResponse.getPrivacyGroupId();
-    } catch (Exception e) {
-      LOG.error("Failed to fetch transaction from Enclave with error " + e.getMessage());
-      return new JsonRpcSuccessResponse(
-          request.getId(), JsonRpcError.PRIVATE_TRANSACTION_RECEIPT_ERROR);
+    } catch (final EnclaveException e) {
+      if (e.getMessage() != null && e.getMessage().equals("EnclavePayloadNotFound")) {
```

Comment:
Using the message in an exception as the indicator of a given problem can be dangerous - the message is really meant for human consumption - if you need to do something like this, its probably better if the exception contained an error code explicitly defining the failure.

Reply:
yes - I have used the correct converter here